core/board.py

```diff
@@ -27,11 +25,21 @@ def update(self, user_shot_choice, is_hit):
                 'I': 8,
                 'J': 9,
                 }
+
+    def update(self, user_shot_choice, is_hit):
         user_letter = user_shot_choice[0]
         user_num = user_shot_choice[1:]
         
-        self.state[rows[user_num]][columns[user_letter]] = 'Hit' if is_hit else 'Miss' 
-            
+        if is_hit == 'Hit':
+            self.state[self.rows[user_num]][self.columns[user_letter]] = 'Hit' 
+        elif is_hit == 'Miss': 
+            self.state[self.rows[user_num]][self.columns[user_letter]] = 'Miss' 
+        else:
+            for each_hit in self.state[self.rows[user_num]][self.columns[user_letter]]['hit_locations']:
```

Comment:
this line took me a while to understand. I also noticed that this is the 3rd time you're using the super long `self.state[self.rows[user_num]][self.columns[user_letter]]` in this method. might be worth consolidating that into a descriptive variable.